src/main/scala-3/com/typesafe/sbt/packager/PluginCompat.scala

```diff
@@ -0,0 +1,58 @@
+package com.typesafe.sbt.packager
+
+import java.io.File
+import java.nio.file.{Path => NioPath}
+import java.util.jar.Attributes
+import sbt.*
+import xsbti.{FileConverter, HashedVirtualFileRef, VirtualFile, VirtualFileRef}
+import sbt.internal.RemoteCache
+
+object PluginCompat:
+  type FileRef = HashedVirtualFileRef
+  type ArtifactPath = VirtualFileRef
+  type Out = VirtualFile
+  type IncludeArtifact = Any => Boolean
+
+  val artifactStr = Keys.artifactStr
+  val moduleIDStr = Keys.moduleIDStr
+  def parseModuleIDStrAttribute(str: String): ModuleID =
+    Classpaths.moduleIdJsonKeyFormat.read(str)
+  def moduleIDToStr(m: ModuleID): String =
+    Classpaths.moduleIdJsonKeyFormat.write(m)
+
+  def parseArtifactStrAttribute(str: String): Artifact =
+    import sbt.librarymanagement.LibraryManagementCodec.ArtifactFormat
+    import sjsonnew.support.scalajson.unsafe.*
+    Converter.fromJsonUnsafe[Artifact](Parser.parseUnsafe(str))
+  def artifactToStr(art: Artifact): String =
+    import sbt.librarymanagement.LibraryManagementCodec.ArtifactFormat
+    import sjsonnew.support.scalajson.unsafe.*
+    CompactPrinter(Converter.toJsonUnsafe(art))
+
+  def toNioPath(a: Attributed[HashedVirtualFileRef])(using conv: FileConverter): NioPath =
+    conv.toPath(a.data)
+  def toNioPath(ref: HashedVirtualFileRef)(using conv: FileConverter): NioPath =
+    conv.toPath(ref)
+  inline def toFile(a: Attributed[HashedVirtualFileRef])(using conv: FileConverter): File =
+    toNioPath(a).toFile()
+  inline def toFile(ref: HashedVirtualFileRef)(using conv: FileConverter): File =
+    toNioPath(ref).toFile()
```

Comment:
I assume this is a performance optimiziation. Does this make a difference here?

Reply:
Removed inline in 2a20c607c0b476feff5d959f06983d243e6f0922